src/Xamarin.Android.Build.Tasks/Tasks/GeneratePackageManagerJava.cs

```diff
@@ -265,7 +265,9 @@ void AddEnvironment ()
 			HashSet<string> archAssemblyNames = null;
 			HashSet<string> uniqueAssemblyNames = new HashSet<string> (StringComparer.OrdinalIgnoreCase);
 			Action<ITaskItem> updateAssemblyCount = (ITaskItem assembly) => {
-				string assemblyName = Path.GetFileName (assembly.ItemSpec);
+				// We need to use the 'RelativePath' metadata, if found, because it will give us the correct path for satellite assemblies - with the culture in the path.
+				string? relativePath = assembly.GetMetadata ("RelativePath");
```

Comment:
Do we still need to change this to use `%(DestinationSubPath)`? I thought we saw that architecture-specific assemblies could also hit a similar issue?

Reply:
No, architecture specific assemblies are handled correctly (if they didn't, we'd see this issue much earlier). `DestinationSubPath` would only made matters worse - because I would have to know which is arch-specific assembly and which is a satellite assembly, to treat the prefix differently. `RelativePath` appears to be unambiguous for this purpose.